main/src/main/java/org/mobilitydata/gtfsvalidator/validator/TranslationConditionalRequirementsValidator.java

```diff
@@ -0,0 +1,119 @@
+
+import javax.inject.Inject;
+import org.apache.commons.lang3.StringUtils;
+import org.mobilitydata.gtfsvalidator.annotation.GtfsValidator;
+import org.mobilitydata.gtfsvalidator.notice.NoticeContainer;
+import org.mobilitydata.gtfsvalidator.notice.TranslationFieldValueDefinedNotice;
+import org.mobilitydata.gtfsvalidator.notice.TranslationFieldValueNotDefinedNotice;
+import org.mobilitydata.gtfsvalidator.notice.TranslationRecordIdDefinedNotice;
+import org.mobilitydata.gtfsvalidator.notice.TranslationRecordIdNotDefinedNotice;
+import org.mobilitydata.gtfsvalidator.notice.TranslationRecordSubIdDefinedNotice;
+import org.mobilitydata.gtfsvalidator.notice.TranslationRecordSubIdNotDefinedNotice;
+import org.mobilitydata.gtfsvalidator.table.GtfsFeedInfoTableLoader;
+import org.mobilitydata.gtfsvalidator.table.GtfsStopTimeTableLoader;
+import org.mobilitydata.gtfsvalidator.table.GtfsTranslation;
+import org.mobilitydata.gtfsvalidator.table.GtfsTranslationTableContainer;
+/**
+ * Validates conditional requirements on `translations.record_id`, `translations.record_sub_id`, and
+ * `translations.field_value` fields:
+ *
+ * <p>`translations.record_id` is: forbidden if `translations.table_name` is `feed_info`, forbidden
+ * if `translations.field_value` is defined, required if `translations.field_value` is empty.
+ *
+ * <p>`translations.record_sub_id` is: forbidden if `translations.table_name` is `feed_info`,
+ * forbidden if `translations.field_value` is defined, required if `translations.table_name` is
+ * `stop_times` and `translations.record_id` is defined.
+ *
+ * <p>`translations.field_value` is: forbidden if `translations.table_name` is `feed_info`,
+ * forbidden if `translations.record_id` is defined, required if `translations.record_id` is empty.
+ *
+ * <p>Generated notices:
+ *
+ * <ul>
+ *   <li>{@link TranslationRecordIdDefinedNotice} - `translations.record_id` is defined when it
+ *       should not be
+ *   <li>{@link TranslationRecordIdNotDefinedNotice} - `translations.record_id` is not defined when
+ *       it should be
+ *   <li>{@link TranslationRecordSubIdDefinedNotice} - `translations.record_sub_id` is defined when
+ *       it should not be
+ *   <li>{@link TranslationRecordSubIdNotDefinedNotice} - `translations.record_sub_id` is not
+ *       defined when it should be
+ *   <li>{@link TranslationFieldValueDefinedNotice} - `translations.field_value` is defined when it
+ *       should be
+ *   <li>{@link TranslationFieldValueNotDefinedNotice} - `translations.field_value` is defined when
+ *       it should not be
+ * </ul>
+ */
+@GtfsValidator
+public class TranslationConditionalRequirementsValidator extends FileValidator {
+
+  private static final String TXT_SUFFIX = ".txt";
+  @Inject GtfsTranslationTableContainer translationTable;
+
+  @Override
+  public void validate(NoticeContainer noticeContainer) {
+    for (GtfsTranslation translation : translationTable.getEntities()) {
+      if (translation
```

Comment:
Unfortunately, that approach does not scale well. GTFS spec defines a lot of files and users may add their extensions with new fields and fields.

We need changes to the validator core. The annotation processor should provide information about the schema and validator classes may receive that as a part of validation context.

I can make the necessary changes to the core when I come back after 15 March.